content/tutorials/build-apps/build-a-desktop-wallet-in-javascript.md

```diff
@@ -0,0 +1,1460 @@
+    await client.request({
+      "command": "subscribe",
+      "streams": ["ledger"],
+      "accounts": [address]
+    })
+
+    // Reference: https://xrpl.org/subscribe.html#ledger-stream
+    client.on("ledgerClosed", async (rawLedgerData) => {
+      reserve = prepareReserve(rawLedgerData)
+      const ledger = prepareLedgerData(rawLedgerData)
+      appWindow.webContents.send('update-ledger-data', ledger)
+    })
+
+    // Initial Ledger Request -> Get account details on startup
+    // Reference: https://xrpl.org/ledger.html
+    const ledgerResponse = await client.request({
+      "command": "ledger"
+    })
+    const initialLedgerData = prepareLedgerData(ledgerResponse.result.closed.ledger)
+    appWindow.webContents.send('update-ledger-data', initialLedgerData)
+
+    // Reference: https://xrpl.org/subscribe.html#transaction-streams
+    client.on("transaction", async (transaction) => {
+      // Reference: https://xrpl.org/account_info.html
+      const accountInfoRequest = {
+        "command": "account_info",
+        "account": address,
+        "ledger_index": transaction.ledger_index
+      }
+      const accountInfoResponse = await client.request(accountInfoRequest)
+      const accountData = prepareAccountData(accountInfoResponse.result.account_data, reserve)
+      appWindow.webContents.send('update-account-data', accountData)
+    })
+
+    // Initial Account Request -> Get account details on startup
+    // Reference: https://xrpl.org/account_info.html
+    const accountInfoResponse = await client.request({
+      "command": "account_info",
+      "account": address,
+      "ledger_index": "current"
+    })
+    const accountData = prepareAccountData(accountInfoResponse.result.account_data)
+    appWindow.webContents.send('update-account-data', accountData)
+
+    // Step 3 code modifications - end
+  })
+}
+```
+
+As the account we want to query is known only after the user enters an address, we had to wrap our application logic into an event handler:
+
+```javascript
+ipcMain.on('address-entered', async (event, address) =>  {
+  // ...
+})
+```
+
+In addition to the subscription to the ledger stream we also can subscribe the client to specific addresses, and we use this feature here to subscribe to an account address which we are going to prompt the user for:
+
+```javascript
+await client.request({
+  "command": "subscribe",
+  "streams": ["ledger"],
+  "accounts": [address]
+})
+```
+
+After this subscription our code attached listeners to the `ledgerClosed` and the `transactions` event. As soon as a `transaction` event is triggered, we do an `account_info` request to get the latest account status, as a transaction is an operation that changes the accounts state.
+
+In addition to the subscriptions we added each an initial `ledger` and `accountInfo` request to have some data at application startup, otherwise we would see empty fields until something happened on the ledger which would trigger one of our subscriptions.
```

Comment:
nit: wording
```suggestion
In addition to the subscriptions we added an initial `ledger` and `accountInfo` request to have some data at application startup. Otherwise we would see empty fields until something happened on the ledger that triggered one of our subscriptions.
```